programs/nft_api/src/nft_processor.rs

```diff
@@ -0,0 +1,241 @@
+//! Non-fungible token program
+
+use crate::{
+    nft_instruction::{InitializeAccountParams, NftError, NftInstruction},
+    nft_state::NftState,
+};
+use solana_sdk::{
+    account::KeyedAccount,
+    instruction::InstructionError,
+    instruction_processor_utils::{limited_deserialize, next_keyed_account},
+    pubkey::Pubkey,
+};
+
+fn initialize_account(
+    nft_state: &mut NftState,
+    params: InitializeAccountParams,
+    issuer_keyed_account: &KeyedAccount,
+) -> Result<(), InstructionError> {
+    if issuer_keyed_account.signer_key().is_none() {
+        return Err(InstructionError::MissingRequiredSignature);
+    }
+    if nft_state.owner_pubkey != Pubkey::default() {
+        return Err(InstructionError::AccountAlreadyInitialized);
+    }
+    nft_state.owner_pubkey = params.owner_pubkey;
+    nft_state.issuer_pubkey = params.issuer_pubkey;
+    nft_state.id = params.id;
+    Ok(())
+}
+
+fn set_owner(
```

Comment:
My first draft of this code actually called it that, but changed it to `set_owner` because I saw that in an ERC-721 doc and thought, yes, that's all the function does. One transfers from me to you by setting the owner to you.  But one also could *claim* ownership, by being sent the private key corresponding to `old_owner_pubkey` and then setting the owner to oneself.  Calling the function `set_owner` is precise regardless of whether the use case is to transfer or claim the token.